transformer_engine/jax/__init__.py

```diff
@@ -3,32 +3,54 @@
 # See LICENSE for license information.
 """Transformer Engine bindings for JAX"""
 
-from . import flax
-from .fp8 import fp8_autocast, update_collections, update_fp8_metas, get_delayed_scaling
-from .fp8 import NVTE_FP8_COLLECTION_NAME
-from .sharding import MeshResource
-from .sharding import MajorShardingType, ShardingResource, ShardingType
-
-from ..common.utils import deprecate_wrapper
-from ..common.utils import DeprecatedEnum
-
-MajorShardingType = DeprecatedEnum(MajorShardingType,
-                                   "MajorShardingType is deprecating in the near feature.")
-ShardingType = DeprecatedEnum(ShardingType, "ShardingType is deprecating in the near feature.")
-ShardingResource = deprecate_wrapper(
-    ShardingResource,
-    "ShardingResource is renamed to MeshResource, and will be removed in the near feature.")
-
-__all__ = [
-    'NVTE_FP8_COLLECTION_NAME',
-    'fp8_autocast',
-    'update_collections',
-    'update_fp8_metas',
-    'get_delayed_scaling',
-    'MeshResource',
-    'MajorShardingType',
-    'ShardingResource',
-    'ShardingType',
-    'flax',
-    'praxis',
-]
+import ctypes
+import os
+
+from transformer_engine.common import get_te_path
+from transformer_engine.common import _get_sys_extension
+
+
+def _load_library():
+    """Load shared library with Transformer Engine C extensions"""
+    extension = _get_sys_extension()
+    try:
+        so_dir = get_te_path() / "transformer_engine"
+        so_path = next(so_dir.glob(f"transformer_engine_jax.*.{extension}"))
+    except StopIteration:
+        so_dir = get_te_path()
+        so_path = next(so_dir.glob(f"transformer_engine_jax.*.{extension}"))
+
+    return ctypes.CDLL(so_path, mode=ctypes.RTLD_GLOBAL)
+
+
+if "NVTE_PROJECT_BUILDING" not in os.environ:
```

Comment:
Suppressing submodule imports based on an envvar feels very hacky to me. I see that we are doing this because `setup.py` needs to import a helper submodule (`build_tools`), which triggers an import of `__init__.py` before anything has been installed. An alternative approach would be [using importlib to directly load the helper submodule](https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly). It's hacky, but no more than the existing approach. It also removes the need to copy `build_tools` into the current dir and it keeps the messiness confined to `setup.py` (which is not accessible in downstream code).

Reply:
This is a good practice recommendation from @DEKHTIARJonathan

Reply:
For completeness, we still always need to copy build_tools into current directory for a standalone build of sdist.